twisted/ldaptor
# Fix non str vs bytes tests for Python3 porting.

Scope
=====

Another one for #55 

This should be the last opportunistic porting to Python3.
It tried to handle all the remaining test failures which are not due to str vs bytes differences between py2 and py3

For now, I am focusing the porting for python3.5.. once 3.5 is done we can look at other versions.

Changes
=======

I have added a new travis-ci env to run tests on py3.5 and make sure we don't add regressions.

Some tests have explicit bytes markup for strings but they still fail... they will be fixed in follow up PR.

I did some drive by fixes for marking bytes while reading the code and working on fixing other tests.
As long as coverage is 100% i hope that this is ok.
I don't have much time, and I prefer to do the changes now, rather than revert them and re-read the code and check it later.

This does a quick porting of distinguishedname, but after this is merged I plan to have a dedicated PR for doing a proper porting of distinguishedname . Once this is done more tests will pass.

With this PR we have 400 passing tests in comparison with  306 which are in master or a total of 522 tests

For comparison with unknown types, I went for Py3 behaviour and raise TypeError.

Other infra changes:

* coveralls was removed
* A new py35-ported tox env was added to help preventing py3 regresssion for the code which was already green

How to test
=========

Check that changes make sense.

Check the new tests, as they might not assert the right thing.
Some code did not had docstring so I wrote the test based on my understanding of how I think that it should behave.
